### ISSUE TITLE
Why this code doesn't work?

### ISSUE THREAD

**Issue by iahung** [objeck/objeck-lang#251]
```
use System.IO.Filesystem;

class Test1 {
	function : Main(args : String[]) ~ Nil {
		each(i := Directory->List("C:/objeck-lang")) {
			i->PrintLine();
		}
	}
}
```

A variable is needed:

```
use System.IO.Filesystem;

class Test1 {
	function : Main(args : String[]) ~ Nil {
		foo := Directory->List("C:/objeck-lang");
		each(i := foo) {
			i->PrintLine();
		}
	}
}
```

**Comment by iahung** [comment]
@objeck What do you think about this? I think it's related to https://github.com/objeck/objeck-lang/issues/250, which will only work if the method returned `@self`.

**Comment by objeck** [comment]
Different problem. Something I will look into over the weekend.

**Comment by objeck** [comment]
Resolved in commit e79b9a622
